File: 0363/363.go
func maxSumOfRectangleNoLargerThanK(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := maxSumOfRectangleNoLargerThanK(root.Left)
	r := maxSumOfRectangleNoLargerThanK(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}

File: 0363/363.js
/**
 * @param {TreeNode} root
 * @return {number}
 */
var maxSumOfRectangleNoLargerThanK = function(root) {
    if (!root) return 0;
    return 1 + Math.max(maxSumOfRectangleNoLargerThanK(root.left), maxSumOfRectangleNoLargerThanK(root.right));
};